src/nodes.coffee

```diff
@@ -4573,6 +4573,53 @@ exports.Switch = class Switch extends Base
     fragments.push @makeCode @tab + '}'
     fragments
 
+  astType: -> 'SwitchStatement'
+
+  casesAst: (o) ->
```

Comment:
Switch is a pretty interesting AST transformation

We want to expand our multi-`when` cases into multiple `SwitchCase` AST nodes (so basically the structure of the `switch` statement that we output in JS) for compatibility/reuse with JS switch ASTs

Eg this:
```
switch yes
  when a, b then c
```
should generate an AST like:
```
type: 'SwitchStatement'
cases: [
  type: 'SwitchCase'
  test:
    type: 'Identifier'
    name: 'a'
  consequent: []
,
  type: 'SwitchCase'
  test:
    type: 'Identifier'
    name: 'b'
  consequent: [
    type: 'ExpressionStatement'
    expression:
      type: 'Identifier'
      name: 'c'
  ]
]
```
which is the same AST generated for this JS:
```
switch (true) {
  case a:
  case b:
    c;
}
```

So it's not a one-to-one mapping between our `SwitchCase`s and the AST `SwitchCase`s. I didn't see a great way to do that mapping into our node classes (which would allow us to use the typical call-`ast()`-on-the-children approach) so here there's a lot of manual assembling of the AST objects and their location data